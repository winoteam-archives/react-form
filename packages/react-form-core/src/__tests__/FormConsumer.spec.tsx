import React from 'react'
import { Form, FormConsumer, FormField } from './../'
import renderer from 'react-test-renderer'

describe('<Form />', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Form initialValues={{ name: 'Léo' }} onSubmit={() => void 0}>
        {() => (
          <FormConsumer>
            {form => <div>{JSON.stringify(form)}</div>}
          </FormConsumer>
        )}
      </Form>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('re-renders correctly', () => {
    let onChange: (value: any) => void = () => void 0
    const renderSpy = jest.fn()
    renderer.create(
      <Form initialValues={{ name: 'Léo' }} onSubmit={() => void 0}>
        {() => (
          <div>
            <FormField name="name">
              {field => {
                onChange = field.onChange
                return null
              }}
            </FormField>
            <FormConsumer>
              {form => {
                renderSpy(form)
                return <div>{JSON.stringify(form)}</div>
              }}
            </FormConsumer>
          </div>
        )}
      </Form>,
    )
    onChange('LéLé')
    expect(renderSpy).toHaveBeenCalledTimes(2)
  })
})
