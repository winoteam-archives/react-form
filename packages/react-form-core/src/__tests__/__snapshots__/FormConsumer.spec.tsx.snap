// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders correctly 1`] = `
<div>
  {"values":{"name":"LÃ©o"},"errors":{},"touched":{},"isSubmitting":false,"submitCount":0}
</div>
`;
